packages/core/src/render3/i18n/i18n_apply.ts

```diff
@@ -198,7 +200,10 @@ export function applyMutableOpCodes(
               attrValue, null);
           break;
         default:
-          throw new Error(`Unable to determine the type of mutate operation for "${opCode}"`);
+          const errorMessage = typeof ngDevMode === 'undefined' || ngDevMode ?
+              `Unable to determine the type of mutate operation for "${opCode}"` :
+              '';
+          throw new RuntimeError(RuntimeErrorCode.MUTABLE_OPCODES, errorMessage);
```

Comment:
In this code we can call `ngDevMode` without extra checks (see some usages above). We could also wrap the entire throw into `ngDevMode` check (it's ok to tree-shake away this in prod bundles):
```suggestion
          if (ngDevMode) {
            throw new RuntimeError(
              RuntimeErrorCode.INVALID_I18N_STRUCTURE,
              `Unable to determine the type of mutate operation for "${opCode}"`);
          }
```